oap-server/oal-rt/src/main/java/org/apache/skywalking/oal/rt/OALRuntime.java

```diff
@@ -315,6 +326,30 @@ private Class generateMetricsClass(AnalysisResult metricsStmt) throws OALCompile
         streamAnnotation.addMemberValue("processor", new ClassMemberValue(METRICS_STREAM_PROCESSOR, constPool));
 
         annotationsAttribute.addAnnotation(streamAnnotation);
+
+        // Add @BanyanDB.TopNAggregation if applicable
+        // 1. groupByTagName is not empty
+        // 2. parentClass implements LongValueHolder, IntValueHolder or DoubleValueHolder
+        try {
+            if (Arrays.stream(parentMetricsClass.getInterfaces())
+                    .anyMatch(i -> i.getSimpleName().equals("IntValueHolder") ||
+                            i.getSimpleName().equals("LongValueHolder") ||
+                            i.getSimpleName().equals("DoubleValueHolder"))) {
```

Comment:
I think `isAssignableFrom` is a better way.

Reply:
For `CtClass` in Javassist, it that possible to use `isAssignableFrom`?

Reply:
Oops, forgot this. Then, use the full class name, such as `LongValueHolder.class.getCanonicalName()`?